main.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+    "encoding/binary"
+    "fmt"
+    "os"
+)
+
+type TCPHeader struct {
+    SourcePort      uint16
+    DestinationPort uint16
+    SequenceNumber  uint32
+    AckNumber       uint32
+    DataOffset      uint8
+    Reserved        uint8
+    Flags           uint16
+    WindowSize      uint16
+    Checksum        uint16
+    UrgentPointer   uint16
+}
+
+func main() {
+    file, err := os.Open("data.bin")
+    if err != nil {
+        fmt.Println("Error opening file:", err)
+        return
+    }
+    defer file.Close()
+
+    // Read binary data from file
+    data := make([]byte, 24)
+    _, err = file.Read(data)
+    if err != nil {
+        fmt.Println("Error reading file:", err)
+        return
+    }
+
+    // Parse TCP header
+    header := TCPHeader{}
+    err = binary.Read(bytes.NewReader(data), binary.BigEndian, &header)
```

Comment:
extract this part as a standalone function; provide unit tests on the new function